openssl-dynamic/src/main/c/ssl.c

```diff
@@ -653,20 +633,13 @@ static void ssl_set_thread_id(CRYPTO_THREADID *id)
     CRYPTO_THREADID_set_numeric(id, ssl_thread_id());
 }
 
-static apr_status_t ssl_thread_cleanup(void *data)
+static void ssl_thread_cleanup()
 {
-    UNREFERENCED(data);
     CRYPTO_set_locking_callback(NULL);
     CRYPTO_THREADID_set_callback(NULL);
     CRYPTO_set_dynlock_create_callback(NULL);
```

Comment:
also be sure to use the `ssl_dyn_destroy_function` (should be accessible via `CRYPTO_get_dynlock_destroy_callback`) on the result of `CRYPTO_get_dynlock_create_callback`.